sagan/models/spectral_norm.py

```diff
@@ -0,0 +1,20 @@
+import tensorflow as tf
+from tensorflow import keras
+
+
+class SpectralNorm(keras.constraints.Constraint):
```

Comment:
I recommend that instead of trying to implement the spectral norm from scratch, we use [`tfa.layers.SpectralNormalization`](https://www.tensorflow.org/addons/api_docs/python/tfa/layers/SpectralNormalization).